pkg/client/zfslocalpv.go

```diff
@@ -91,6 +91,51 @@ func (k K8sClient) GetZFSVols(volNames []string, rType util.ReturnType, labelSel
 }
 
 // GetZFSNodes return a list of ZFSNodes
-func (k K8sClient) GetZFSNodes() (*zfs.ZFSNodeList, error) {
-	return k.ZFCS.ZfsV1().ZFSNodes("").List(context.TODO(), metav1.ListOptions{})
+func (k K8sClient) GetZFSNodes(volNames []string, rType util.ReturnType, labelSelector string, options util.MapOptions) (*zfs.ZFSNodeList, map[string]zfs.ZFSNode, error) {
+	zfsNode, err := k.ZFCS.ZfsV1().ZFSNodes("").List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		return nil, nil, err
+	}
+	var list []zfs.ZFSNode
+	if len(volNames) == 0 {
+		list = zfsNode.Items
+	} else {
+		zvsMap := make(map[string]zfs.ZFSNode)
+		for _, zn := range zfsNode.Items {
+			zvsMap[zn.Name] = zn
+		}
+		for _, name := range volNames {
+			if zv, ok := zvsMap[name]; ok {
+				list = append(list, zv)
+			} else {
+				// This might be omitted
+				fmt.Printf("Error from server (NotFound): zfsVolume %s not found\n", name)
```

Comment:
[Enhancement] This will basically require another round of refactors, it could be returning a slice of errors or appending like 2 non nil errors and returning it to the caller stating that out of 10 nodes 2 aren't found.